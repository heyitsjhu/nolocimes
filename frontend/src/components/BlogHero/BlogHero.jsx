import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';

const useStyles = makeStyles(theme => ({
  blogHeroComponent: {
    width: '100%',
    height: '50vh',
    maxHeight: 480,
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
  },
}));

export default ({ srcUrl }) => {
  const classes = useStyles();
  const imgUrl = srcUrl ? srcUrl : process.env.PUBLIC_URL + '/static/images/blogHeroImage.jpg';

  return (
    <Box className={classes.blogHeroComponent} style={{ backgroundImage: `url('${imgUrl}')` }}>
      {/* {props.title && <Typography variant="h5">{props.title}</Typography>} */}
    </Box>
  );
};
