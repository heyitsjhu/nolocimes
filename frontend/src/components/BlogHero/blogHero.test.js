import React from 'react';
import renderer from 'react-test-renderer';
import pretty from 'pretty';
import { unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import BlogHero from './BlogHero';

let container = null;

describe('[components] - BlogHero', () => {
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });
  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test('it renders without failing snapshot', () => {
    let tree = renderer.create(<BlogHero />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
