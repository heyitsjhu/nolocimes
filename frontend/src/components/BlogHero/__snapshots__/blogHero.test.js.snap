// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components] - BlogHero it renders without failing snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 makeStyles-jottingPadHeroComponent-1"
  style={
    Object {
      "backgroundImage": "url('')",
    }
  }
/>
`;
